app/main.py

```diff
@@ -1,34 +1,56 @@
+from typing import Optional
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
+        current_point = [start[0], start[1]]
+        while current_point[0] != end[0]:
+            self.decks.append(Deck(current_point[0], current_point[1]))
+            current_point[0] += 1
+        while current_point[1] != end[1]:
+            self.decks.append(Deck(current_point[0], current_point[1]))
+            current_point[1] += 1
+        self.decks.append(Deck(current_point[0], current_point[1]))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Optional[Deck]:
+        for deck in self.decks:
+            if (row, column) == (deck.row, deck.column):
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> str:
+        current_deck = self.get_deck(row, column)
+        if current_deck.is_alive:
```

Comment:
```suggestion
        if current_deck is not None and current_deck.is_alive:
```

Reply:
after all actions in class Battleship it looks like impossible to get None from current_deck()